### DISCUSSION TITLE
Is it necessary to restart Spacemacs to install packages?

### DISCUSSION THREAD

**Discussion by Kazark** [syl20bnr/spacemacs#14782] [Q&A] [answered]
After `SPC f e U`, a message is printed advising:

> --> performing backup of package(s) to update...
> --> preparing update of package yaml-mode... [46/46]
> --> 46 package(s) to be updated.
> 
> Restart Emacs to install the updated packages. (SPC q r)

However I found accidentally that doing `SPC f e d` (open `.spacemacs`) `SPC f e R` installed the packages, though with an error:

> Restart Emacs to install the updated packages. (SPC q r)
> Error in dotspacemacs/user-init: Wrong type argument: symbolp, 22
> Found 44 new package(s) to install...
> --> installing package: yaml-mode@yaml... [44/44]

(Technically I also added `evil-better-jumper` layer before `SPC f e R`.)

What kind of state am I in? Do I need a restart?

**Comment by lebensterben** [comment]
> Error in dotspacemacs/user-init: Wrong type argument: symbolp, 22

Similar bug should have been reported and it happened to me also.
Probably error code 213 but I'm not sure.
Anyway 22 is not a printable character, but a control character. And this is a bug. I've tried to debug this but failed.

**Comment by pataquets** [comment, marked as answer]
If you want Spacemacs to do it, I'm afraid it is.
However, if you want to just test some random package quickly, you can use Emacs' `package-install` command an try it right away. Later, if you decide to keep it, don't forget to add it to Spacemacs' `init.el` or it will be deleted ([docs](https://develop.spacemacs.org/doc/FAQ.html#install-a-package-not-provided-by-a-layer))
